cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -228,7 +228,8 @@ class parquet_field_string : public parquet_field {
  * @return True if field types mismatch or if the process of reading a
  * string fails
  */
-struct parquet_field_string_list : public parquet_field_list<std::string, FieldType::BINARY> {
+class parquet_field_string_list : public parquet_field_list<std::string, FieldType::BINARY> {
```

Comment:
ELI5:  Why was this change required?  Isn't the old version a valid equivalent?

Is it because of (the equivalent of) the [Google Style Guide](https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes) stating that `structs` need to be POD/passive objects?

Reply:
No, this was necessary because the header file declares this object as a class instead of a struct. We could equivalently have changed the header to use struct instead, but the two should be aligned because under certain compilers (Windows) the results could be ABI-incompatible. See https://clang.llvm.org/docs/DiagnosticsReference.html#wmismatched-tags